tests/unit/app/services/auth.test.js

```diff
@@ -0,0 +1,147 @@
+import axios from 'axios'
+import Auth from 'app/js/services/auth.js'
+import Cookies from 'js-cookie'
+import { Promise } from 'es6-promise'
+
+describe('Auth', () => {
+
+  beforeEach(() => {
+    jasmine.Ajax.install();
+  });
+
+  afterEach(() => {
+    jasmine.Ajax.uninstall();
+  });
+
+  describe('init', () => {
+    it('call initSession', () => {
+      spyOn(Auth, 'initSession');
+      Auth.init();
+      expect(Auth.initSession).toHaveBeenCalled();
+    })
+
+    it('call addXHRInterceptor', () => {
+      spyOn(Auth, 'addXHRInterceptor');
+      Auth.init();
+      expect(Auth.addXHRInterceptor).toHaveBeenCalled();
+    })
+  })
+
+  describe('addXHRInterceptor', () => {
+    beforeEach(()=>{
+      Auth.addXHRInterceptor();
+    })
+
+    it('add authorization cookie to ajax request', (done)=>{
+      Cookies.set('accessToken', 'secretToken')
+      jasmine.Ajax.stubRequest('/foo').andReturn({status: 200});
+
+      axios('/foo').then(() => {
+        let request = jasmine.Ajax.requests.mostRecent();
+        expect(request.requestHeaders.Authorization).toEqual('secretToken');
+        done();
+      });
+    })
+
+    it('catch 401 and call resolveRejectedRequest', (done)=>{
+      jasmine.Ajax.stubRequest('/foo').andReturn({status: 401});
+
+      spyOn(Auth, 'resolveRejectedRequest').and.returnValue({})
+
+      axios('/foo').then(()=>{
+        expect(Auth.resolveRejectedRequest).toHaveBeenCalled();
+        done();
+      });
+    })
+  })
+
+  describe('initSession', () => {
+    it('call client portal admin and check that session is valid', () => {
+      spyOn(axios, 'get');
+      Auth.initSession();
+      expect(axios.get).toHaveBeenCalledWith('http://test.com/api/v1/session');
+    })
+  })
+
+  describe('redirectToLoginPage', () => {
+    it('call routeUtils.redirectTo with login url', () => {
+      let routeUtils = { redirectTo: jasmine.createSpy('redirectTo') }
+      let AuthRewireAPI = Auth.__RewireAPI__
+      AuthRewireAPI.__Rewire__('routeUtils', routeUtils)
+
+      Auth.redirectToLoginPage()
+
+      expect(routeUtils.redirectTo).toHaveBeenCalledWith(
+        'http://test.com/#/auth/sign_in?return_url=http://test.com/analytics'
+      );
+    })
+
+  })
+
+  describe('resolveRejectedRequest', () => {
+
+    beforeEach(() => {
+      Auth.addXHRInterceptor()
+
+      Cookies.set('refreshToken', 'secretRefreshToken')
+      jasmine.Ajax.stubRequest('/unauthorized').andReturn({status: 401});
+      jasmine.Ajax.stubRequest('http://test.com/api/v1/api_session/refresh').andReturn({
+        responseText: "{\"access_token\":\"NewAccessToken\"}"
+      });
+
+      spyOn(axios, 'post').and.callThrough()
```

Comment:
For now test like this is ok, but as the application becomes bigger we will need a layer to isolate the API calls, so instead of intercepting the AJAX request we can intercept that service. Easier to mock/fake and we are not coupling the tests with the library that we are using (axios in this case). But that can be done in a quality day or in a near future, so don't worry too much.